expense/expenseController.js

```diff
@@ -0,0 +1,35 @@
+const express = require("express");
+const router = express.Router();
+
+const Expense = require("./expenseModel.js");
+
+router
+  .route("/")
+  .post((req, res) => {
```

Comment:
Here's an example of _saving_ refs to an instance of an Expenses object. The idea is to pull the ObjectId from Budget and Category DB and saving it as refs to Expenses. You'll have to add `.catch()` methods to handle errors.

```
.post((req, res) => {
  const { amount, budget, category, description } = req.body
  if (amount && description && budget && category) {
    Category.findOne({ title: category }).then(cat => {
      Budget.findOne({ title: budget }).then(bud => {
        const expense = new Expense({
          amount,
          budget: bud._id,
          category: cat._id,
          description
        })
        expense.save().then(item => {
          res.json(item)
        })
      })
    })
```



Make sure your POST `req.body` has:
```
{
    "description": "description",
    "amount": 210,
    "budget": "aaa",
    "category": "Food/Dining"
}
```
And that `'aaa'` and `'Food/Dining'` exist in Budget and Category DB, respectively